source/common/config/utility.cc

```
@@ -81,20 +81,27 @@ void Utility::checkFilesystemSubscriptionBackingPath(const std::string& path) {
 void Utility::checkApiConfigSourceSubscriptionBackingCluster(
     const Upstream::ClusterManager::ClusterInfoMap& clusters,
     const envoy::api::v2::core::ApiConfigSource& api_config_source) {
-  if (api_config_source.cluster_names().size() != 1) {
-    // TODO(htuch): Add support for multiple clusters, #1170.
-    throw EnvoyException(
-        "envoy::api::v2::core::ConfigSource must have a singleton cluster name specified");
-  }
+  if (api_config_source.api_type() == envoy::api::v2::core::ApiConfigSource::GRPC) {
+    if (api_config_source.cluster_names().size() != 0) {
```

We should also throw an exception if `grpc_services().size() != 1`.